src/panels/config/automation/action/types/ha-automation-action-delay.ts

```diff
@@ -17,22 +22,56 @@ export class HaDelayAction extends LitElement implements ActionElement {
   }
 
   protected render() {
-    const { delay } = this.action;
+    let data: HaFormTimeData = {};
+
+    if (typeof this.action.delay !== "object") {
+      const parts = this.action.delay?.toString().split(":") || [];
+      data = {
+        hours: Number(parts[0]),
+        minutes: Number(parts[1]),
+        seconds: Number(parts[2]),
+        milliseconds: Number(parts[3]),
+      };
+    } else {
+      const { days, minutes, seconds, milliseconds } = this.action.delay;
+      let { hours } = this.action.delay || 0;
+      hours = (hours || 0) + (days || 0) * 24;
+      data = {
+        hours: hours,
+        minutes: minutes,
+        seconds: seconds,
+        milliseconds: milliseconds,
+      };
+    }
+
+    const schema: HaFormTimeSchema = {
+      type: "positive_time_period_dict",
+      name: "",
+      enableMillisec: true,
+    };
 
     return html`
-      <paper-input
-        .label=${this.hass.localize(
-          "ui.panel.config.automation.editor.actions.type.delay.delay"
-        )}
-        name="delay"
-        .value=${delay}
+      <ha-form
```

Comment:
We should not use `ha-form` here, you can extract what you need and create a separate component that you share between here and `ha-form`